src/txHandler/Bulker.TxHandler.ts

```diff
@@ -1,15 +1,194 @@
-import { Operation } from "src/simulation/simulation.types";
+import { BigNumber, constants } from "ethers";
+import { deepCopy, getAddress } from "ethers/lib/utils";
+import { Subject } from "rxjs";
+import { TxOperation } from "src/simulation/simulation.types";
 
+import { WadRayMath } from "@morpho-labs/ethers-utils/lib/maths";
+import { maxBN } from "@morpho-labs/ethers-utils/lib/utils";
+
+import { MorphoAaveV3DataHolder } from "../MorphoAaveV3DataHolder";
+import addresses from "../contracts/addresses";
+import CONTRACT_ADDRESSES from "../contracts/addresses";
 import { TransactionOptions } from "../types";
 
+import { Bulker } from "./Bulker.TxHandler.interface";
 import { NotifierManager } from "./NotifierManager";
-import { IBatchTxHandler } from "./TxHandler.interface";
-
-export default class BulkerTxHandler extends NotifierManager implements IBatchTxHandler {
-  public handleBatchTransaction(
-    operations: Operation[],
-    options?: TransactionOptions | undefined
-  ): Promise<any> {
-    return Promise.reject("TODO: to implement");
+import { IBatchTxHandler, ISimpleTxHandler } from "./TxHandler.interface";
+
+import TransactionType = Bulker.TransactionType;
+
+export default class BulkerTxHandler
+  extends NotifierManager
+  implements IBatchTxHandler
+{
+  _operations: TxOperation[] = [];
+  _bulkerTransactions: Bulker.Transactions[] = [];
+  _value = constants.Zero;
+
+  public readonly operations$ = new Subject<TxOperation[]>();
+
+  protected set operations(ops: TxOperation[]) {
+    this._operations = ops;
+    this.operations$.next(ops);
+  }
+
+  public getOperations(): TxOperation[] {
+    return deepCopy(this._operations);
+  }
+  public getValues(): BigNumber {
+    return BigNumber.from(this._value);
+  }
+
+  public getBulkerTransactions(): Bulker.Transactions[] {
+    return deepCopy(this._bulkerTransactions);
+  }
+
+  constructor(protected _dataHolder: MorphoAaveV3DataHolder) {
+    super();
+  }
+
+  addOperation(operation: TxOperation): void {
+    switch (operation.type) {
+      case TransactionType.supply:
+      case TransactionType.supplyCollateral:
+        const { batch, value } = this.validateSupply(
+          operation.underlyingAddress,
+          operation.amount
+        );
+        this._value = this._value.add(value);
+        this._bulkerTransactions = [...this._bulkerTransactions, ...batch];
+        this.operations = [...this._operations, operation];
+        break;
+      default:
+        throw Error("Not implemented");
+    }
+  }
+
+  executeBatch(options: TransactionOptions | undefined): Promise<any> {
+    return Promise.resolve(undefined);
+  }
+
+  removeOperation(index: number): void {
+    if (index < 0 || index >= this._operations.length) {
+      throw new Error("Invalid index");
+    }
+    const newOperations = [...this._operations.filter((_, i) => i !== index)];
+
+    //  TODO:  validate
+    this.operations = newOperations;
```

Comment:
What about the `_bulkerTransactions?`

What I would do is to keep `_bulkerTransactions ` as a `Bulker.Transactions[][]` so that each Operation is associated with a list of `Bulker.Transactions`. One could then flatten the array on execution

Reply:
Or, if we want to keep things simple, just have one array containing object with the user operation and an inner array with the bulker operations.

Reply:
Could do the job as well

Reply:
I don't really have a preference here